docs/DeveloperGuide.md

```diff
@@ -144,92 +139,88 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
-
-</div>
-
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
-
-![UndoRedoState4](images/UndoRedoState4.png)
-
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. Reason: It no longer makes sense to redo the `add n/David …​` command. This is the behavior that most modern desktop applications follow.
-
-![UndoRedoState5](images/UndoRedoState5.png)
-
-The following activity diagram summarizes what happens when a user executes a new command:
-
-![CommitActivityDiagram](images/CommitActivityDiagram.png)
-
-#### Design consideration:
-
-##### Aspect: How undo & redo executes
-
-* **Alternative 1 (current choice):** Saves the entire address book.
-  * Pros: Easy to implement.
-  * Cons: May have performance issues in terms of memory usage.
-
-* **Alternative 2:** Individual command knows how to undo/redo by
-  itself.
-  * Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-  * Cons: We must ensure that the implementation of each individual command are correct.
-
-_{more aspects and alternatives to be added}_
-
-### \[Proposed\] Data archiving
-
-_{Explain here how the data archiving feature will be implemented}_
+### Contact and tag management
+![contact_tag_diagram](images/ContactTagDiagram.png)
+
+##### General Design
+**`Person`** component: 
+
+In Athena, contacts are represented by `Person` objects. `Person` objects have several properties such as email, address, etc. However, a `Person` can also be tagged with multiple `Tag`s.
+- `AddressBook` handles all direct matters concerning `Person` objects. It has a `TagManager` and `UniquePersonList`.  
+- `UniquePersonList` keeps track of all `Person` objects. Using `Person` class' `isSamePerson(Person)` method, it ensures that there are no duplicate contacts in Athena.
+- `TagManager` keeps track of which contacts contain which tags. It uses a hash map, with `Tag`s as keys and stores a set of `Person`s as its value.
+
+All manipulation of `Person` objects have to be done through `AddressBook`. `AddressBook` provides simple methods that can be used by higher-level components such as 
+- `void removePerson(Person)`
+- `void addPerson(Person)`
+- `boolean hasPerson(Person)`
+- and more...
+
+**`Tag`** component:
+
+`Tag`s are represented by a single alphanumeric string with no spaces. In v1.3, support for child-tagging was implemented. This allows directional relations to be established between `Tag`s. Certain commands also affect a `Tag` and its child-tags.
```

Comment:
is it necessary to mention when the support was added?

Reply:
also i think the last sentence sounds nicer as:

"Certain commands that affect a `Tag` may also affect its child-tags." (also maybe use a different word besides affect)